Combine training dataset

sabinevidal opened the pull request:
For task 1 in #108:

Script to combine current training data in kindly_luis_export.json with format `[text], [label_number]`.
Where the label number is determined by whether the intent in the original dataset is 'bullying', or was set to 'ignore' or 'none'.

Original script was not differentiating between the `utterances[index].intent` when it was given in the string format, but worked when intent strings were included in an array. Possibly don't need `bully` variable?

nathanfletcher commented:
> For task 1 in #108:
> 
> Script to combine current training data in kindly_luis_export.json with format `[text], [label_number]`. Where the label number is determined by whether the intent in the original dataset is 'bullying', or was set to 'ignore' or 'none'.
> 
> Original script was not differentiating between the `utterances[index].intent` when it was given in the string format, but worked when intent strings were included in an array. Possibly don't need `bully` variable?

@sabinevidal yes you're absolutely right. We don't need the bully variable from Luis in the JSON

nathanfletcher commented on modeling/dataset/training_data.txt:
```diff
@@ -0,0 +1,62 @@
+[1], [back off]
```
@sabinevidal I checked in with how we would be able to process this easily in python in the future, and the best method would be to remove the square braces `[]`.
So this would end up being 
```0, haha```

instead of 

```[0], [haha]```

nathanfletcher replied:
@sabinevidal that way we could use Pandas easily to process and train. What do you think?

lacabra replied:
+1 to JSON file to account for multiline input setences like:
```
YOU
ARE
K
I
N
D
```

which I am afraid will pose a challenge with the simpler `0, haha` proposed model.

nathanfletcher reviewed (changes requested):
hey @sabinevidal sorry this took a bit long. I've requested your comments on some suggested changes on how the new combined training data file will affect processing later on down the line.
Let me know what you think

lacabra commented:
Thanks Sabine for reverting these changes. This is one valid way to do it. The upside of this approach is that it keeps a linear history, the downside is that it clutters history with unnecessary commits. There are big projects where they don't allow to re-write history, but since this project is small, and we allow it, I would prefer another approach that only includes the relevant commits (I acknowledge this is an opinionated take on the matter). So, for the reasons outlined here, and for educational purposes, can you please try the following (search `force push` for additional context):

1. Identify the last good commit to keep (either by `git log` or looking at the commits of this PR), which is bf5d29d
2. Then run the following to go back to that commit
```bash
git reset --hard bf5d29d
```
3. Force push the chosen commit above as the last valid one, effectively discarding anything afterwards:
```bash
git push --force
```

Let me know if you have any questions, thanks!

lacabra commented:
When doing the above, please make a copy of the changes introduced in the commits we will be discarding elsewhere, so that you can refer to them later and include them in different PRs. Otherwise those changes will be lost forever, as we are erasing any trace of them.

sabinevidal commented:
@lacabra and @nathanfletcher 
Have reverted and removed the changes from this PR (and have created a new branch with the other changes). Thanks for the guidance on this! Understanding the difference between the options to re-write history now 👍

Have also updated the script to combine the contribution data into a json file as discussed above.

lacabra commented on modeling/dataset/luisport_combined.js:
```diff
@@ -0,0 +1,22 @@
+/* This script is what was used to extract data from the kindly_luis_export.json file, combined test and label as '[text], [label_number]'*/
```
please update this comment to reflect the change in formatting from `[text], [label_number]` to
```
    {
        "text": "[text]",
        "intent": [0 or 1]
    },
```

lacabra reviewed (approved):
@sabinevidal this looks fantastic! Thanks for promptly addressing my change requests. I have one more and very minor request, see below. Otherwise, this looks good to me, and I leave it to @nathanfletcher to give the final approval and merge

nathanfletcher reviewed (approved):
Looks good, especially the new format. I'm sure it will not be a problem to process it. In fact it might make it easier to process.
